kivy/core/image/__init__.py

```diff
@@ -278,6 +283,8 @@ def populate(self):
             # set as our current texture
             self._textures.append(texture)
 
+            self._durations.append(self._data[count].duration)
```

Comment:
@matham the data is yielded in im_pil and is stored here, so duration is not empty by default. I'm not sure about the code that tests that durations is not None, because i don't see how the users could have a case of durations==None.

Reply:
> @matham the data is yielded in im_pil and is stored here, so duration is not empty by default. I'm not sure about the code that tests that durations is not None, because i don't see how the users could have a case of durations==None.

We use this in kivy.core.image.Image._img_iterate() if auto_anim_delay is True. So? steps are
create Image with parameters
Image.__init__()
1) if auto_anim_delay == False and durations == None then like before
2) if auto_anim_delay == False and durations then Image.durations = durations from creating parameters
3) if auto_anim_delay == True then when in _img_iterate() we set self.durations = self.image.durations - thiese durations are from GIF file

In case 3) durations from creating parameters are not used.